Userland/Libraries/LibSQL/Value.cpp

```diff
@@ -525,60 +561,60 @@ ResultOr<Value> Value::divide(Value const& other) const
     if (auto double_maybe = to_double(); double_maybe.has_value()) {
         if (auto other_double_maybe = other.to_double(); other_double_maybe.has_value())
             return Value(double_maybe.value() / other_double_maybe.value());
-        if (auto int_maybe = other.to_int(); int_maybe.has_value())
+        if (auto int_maybe = other.to_i64(); int_maybe.has_value())
             return Value(double_maybe.value() / (double)int_maybe.value());
-    } else if (auto int_maybe = to_int(); int_maybe.has_value()) {
+    } else if (auto int_maybe = to_i64(); int_maybe.has_value()) {
         if (auto other_double_maybe = other.to_double(); other_double_maybe.has_value())
             return Value((double)int_maybe.value() / other_double_maybe.value());
-        if (auto other_int_maybe = other.to_int(); other_int_maybe.has_value())
+        if (auto other_int_maybe = other.to_i64(); other_int_maybe.has_value())
             return Value(int_maybe.value() / other_int_maybe.value());
     }
     return invalid_type_for_numeric_operator(AST::BinaryOperator::Division);
 }
 
 ResultOr<Value> Value::modulo(Value const& other) const
 {
-    auto int_maybe_1 = to_int();
-    auto int_maybe_2 = other.to_int();
+    auto int_maybe_1 = to_i64();
+    auto int_maybe_2 = other.to_i64();
     if (!int_maybe_1.has_value() || !int_maybe_2.has_value())
         return invalid_type_for_numeric_operator(AST::BinaryOperator::Modulo);
     return Value(int_maybe_1.value() % int_maybe_2.value());
 }
 
 ResultOr<Value> Value::shift_left(Value const& other) const
 {
-    auto u32_maybe = to_u32();
-    auto num_bytes_maybe = other.to_int();
-    if (!u32_maybe.has_value() || !num_bytes_maybe.has_value())
+    auto u64_maybe = to_u64();
+    auto num_bytes_maybe = other.to_i64();
+    if (!u64_maybe.has_value() || !num_bytes_maybe.has_value())
         return invalid_type_for_numeric_operator(AST::BinaryOperator::ShiftLeft);
-    return Value(u32_maybe.value() << num_bytes_maybe.value());
+    return Value(u64_maybe.value() << num_bytes_maybe.value());
```

Comment:
I know this is not something caused by this PR but this arbitrary shifting is UB and will do weird stuff with your program

Reply:
Yeah this was an existing issue, but we now check that the amount we shift is less than the width of the underlying type (plus do all sorts of other overflow checks).